### ISSUE TITLE
grpc gen proto for php not working, protoc-gen-php: program not found or is not executable

### ISSUE THREAD

**Issue by thorie7912** [grpc/grpc#22108]
I am using ubuntu 18.04 + php 7.2 and following the php tutorial here:

https://grpc.io/docs/tutorials/basic/php/

The step `git clone -b v1.27.0 https://github.com/grpc/grpc` seemed fine.

The step `cd grpc && git submodule update --init && make grpc_php_plugin` seemed to complete ok although there were some C-related notices.

```
[AR]      Creating /home/thorie/work/grpc/libs/opt/libgrpc_plugin_support.a
[HOSTCXX] Compiling src/compiler/php_plugin.cc
In file included from ./src/compiler/config_protobuf.h:33:0,
                 from ./src/compiler/config.h:22,
                 from src/compiler/php_plugin.cc:23:
third_party/protobuf/src/google/protobuf/io/printer.h: In member function ‘virtual void google::protobuf::io::AnnotationCollector::AddAnnotationNew(google::protobuf::io::AnnotationCollector::Annotation&)’:
third_party/protobuf/src/google/protobuf/io/printer.h:68:45: warning: unused parameter ‘a’ [-Wunused-parameter]
   virtual void AddAnnotationNew(Annotation& a) {}
                                             ^
[HOSTLD]  Linking /home/thorie/work/grpc/bins/opt/grpc_php_plugin
```

However, the next step failed.

```
thorie@thorie-ubuntu:~/work/grpc$ cd examples/php/route_guide
thorie@thorie-ubuntu:~/work/grpc/examples/php/route_guide$ ./route_guide_proto_gen.sh
protoc-gen-php: program not found or is not executable
--php_out: protoc-gen-php: Plugin failed with status code 1.
```

I'm not sure if I missed something in the tutorial, but I don't know what to do from here.

This issue is similar to https://github.com/grpc/grpc/issues/12714 (but that one was closed without a resolution).

**Comment by thorie7912** [comment]
I noticed that when searching for `protoc-gen-php`, it seems to be deprecated? https://github.com/bramp/protoc-gen-php

Maybe this tutorial needs to be updated to use the "official compiler" @ https://github.com/protocolbuffers/protobuf/tree/master/php ?

**Comment by thorie7912** [comment]
I compiled protoc version 3.11.4 and modified the command in the `route_guide_proto_gen.sh` script by removing the parameters `--grpc_out=./` and `--plugin=protoc-gen-grpc=./../../../bins/opt/grpc_php_plugin`, which seemed to solve the issue. The PHP files were properly generated in the `Routeguide` folder.

**Comment by thorie7912** [comment]
Since I wasn't getting the RouteGuideClient as expected, I went back and fetched the `master` branch of `grpc` and re-compiled `grpc_php_plugin` from scratch. I suspected it was a version mismatch, per the documentation and the other ticket.

https://github.com/grpc/grpc/tree/master/src/php#protoc-compiler

So, after re-compiling, and using protoc 3.11.4, I was finally able to run the script.

Results after running:

```
-rw-r--r-- 1 thorie thorie 2441 Feb 22 06:59 Feature.php
-rw-r--r-- 1 thorie thorie 2137 Feb 22 06:59 Point.php
-rw-r--r-- 1 thorie thorie 2451 Feb 22 06:59 Rectangle.php
-rw-r--r-- 1 thorie thorie 3448 Feb 22 06:59 RouteGuideClient.php
-rw-r--r-- 1 thorie thorie 2439 Feb 22 06:59 RouteNote.php
-rw-r--r-- 1 thorie thorie 4339 Feb 22 06:59 RouteSummary.php
```

Looks like I'm back in business. I don't know if I could have used a more specific tag instead of `master` here. I think someone more knowledgeable should update the documentation to help verify the version of `protoc` and version of `grpc` that will properly work together so that the tutorial can be completed more smoothly.

**Comment by dfluff** [comment]
I have just been through what feels like the exact same issue.  The documentation now states `git clone -b v1.32.0 https://github.com/grpc/grpc` rather than v1.27.0, but from there the `make grpc_php_plugin` failed and even when I wrongly thought I'd worked around that issue running `route_guide_proto_gen.sh` generated all the files except the `RouteGuide\RouteGuideClient` file.  Since I'm very new to gRPC I spent ages trying to work out why the run script was throwing errors.
When I stumbled across this post I switch the repo to master rather than the specific version and started again.  This time I've actually got the `RouteGuide\RouteGuideClient` so I am at last making progress.
